src/nvim/search.c

```diff
@@ -6017,3 +6017,9 @@ bool search_was_last_used(void)
 {
   return last_idx == 0;
 }
+
+/// @return  true if 'hlsearch' highlight is currently in use.
+bool hlsearch_used(void)
```

Comment:
- if this is about current state, "using" is more intuitive.
- An existing precedent is `using_script`, I guess it's worth being consistent there
- search.c doesn't have  common prefix for function names, otherwise the name would be something like `search_using_hlsearch()`.

```suggestion
bool using_hlsearch(void)
```

Reply:
I don't think using this in `start_search_hl()` reduces much code, as it is still necessary to call `last_pat_prog()` to set `search_hl.rm.regprog = NULL` even if hlsearch is not in use.